src/test/java/com/google/finapp/FinAppIT.java

```diff
@@ -19,68 +19,98 @@
 import com.google.cloud.spanner.DatabaseClient;
-import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.IntegrationTest;
+import com.google.cloud.spanner.IntegrationTestEnv;
 import com.google.cloud.spanner.Key;
 import com.google.cloud.spanner.KeySet;
 import com.google.cloud.spanner.ResultSet;
-import com.google.cloud.spanner.Spanner;
-import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.testing.RemoteSpannerHelper;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
 import java.math.BigDecimal;
 import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
 import java.util.UUID;
-import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.ValueSource;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
 
+@Category(IntegrationTest.class)
 public class FinAppIT {
 
-  private static SpannerDaoInterface JDBCDao;
-  private static SpannerDaoInterface JavaDao;
+  private static SpannerDaoInterface daoJDBC;
+  private static SpannerDaoInterface daoJava;
   private static DatabaseClient databaseClient;
+  private static Database db;
 
-  static {
-    // set all IDs according to emulator setup
-    String spannerProjectId = "test-project";
-    String spannerInstanceId = "test-instance";
-    String spannerDatabaseId = "test-database";
-    JDBCDao = new SpannerDaoJDBCImpl(spannerProjectId, spannerInstanceId, spannerDatabaseId);
-    SpannerOptions spannerOptions = SpannerOptions.getDefaultInstance();
-    Spanner spanner = spannerOptions.toBuilder().build().getService();
-    databaseClient = spanner
-        .getDatabaseClient(DatabaseId.of(spannerProjectId, spannerInstanceId, spannerDatabaseId));
-    JavaDao = new SpannerDaoImpl(databaseClient);
+  @ClassRule
+  public static IntegrationTestEnv env = new IntegrationTestEnv();
+
+  @BeforeClass
+  public static void setup() throws IOException {
+    RemoteSpannerHelper testHelper = env.getTestHelper();
+    db = testHelper
+        .createTestDatabase(extractStatementsFromSDLFile("src/main/java/com/google/finapp/schema.sdl"));
+    daoJDBC = new SpannerDaoJDBCImpl(testHelper.getOptions().getProjectId(),
+        testHelper.getInstanceId().getInstance(), db.getId().getDatabase());
+    databaseClient = testHelper.getDatabaseClient(db);
+    daoJava = new SpannerDaoImpl(databaseClient);
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    db.drop();
   }
 
-  private SpannerDaoInterface getImpl(String impl) {
-    if (impl.equals("java client")) {
-      return JavaDao;
-    } else {
-      return JDBCDao;
+  @Test
+  public void createAccountTest() throws SpannerDaoException {
+    for (SpannerDaoInterface spannerDao : List.of(daoJava, daoJDBC)) {
```

Comment:
Optional: Have you considered to have an env variable like spanner.testenv.use_jdbc and use it inside the test to chose which dao to instantiate?
Then in the `pom.xml` you could have two variants: integration-test-java, integration-test-jdbc that differ on how they set this environment variable.

Reply:
I changed the tests to now utilize environment to choose the spanner dao impl. Setting up separate commands in `pom.xml` can be looked into in the future. 

running `mvn integration-test` now runs the Java client impl
and `mvn -DSPANNER_USE_JDBC=true integration-test` runs the JDBC impl

Reply:
sgtm